bin/extract_api_surface.py

```
@@ -0,0 +1,314 @@
+
+def _find_source_file(pkg_dir: Path, module_name: str) -> Path | None:
+    p = pkg_dir / f"{module_name}.py"
+    if p.exists():
+        return p
+    p = pkg_dir / module_name / "__init__.py"
+    if p.exists():
+        return p
+    return None
+
+
+def _get_top_level_exports(pkg_dir: Path) -> list[str]:
```

Ruff (0.15.7)

[warning] Too many branches (17 > 12)

(PLR0912)